backend/moonstream/routes/users.py

```diff
@@ -0,0 +1,172 @@
+"""
+The Moonstream users HTTP API
+"""
+import logging
+from typing import Any, Dict
+import uuid
+
+from bugout.data import BugoutToken, BugoutUser
+from bugout.exceptions import BugoutResponseException
+from fastapi import (
+    FastAPI,
+    Form,
+    HTTPException,
+    Request,
+)
+from fastapi.middleware.cors import CORSMiddleware
+
+from ..middleware import BroodAuthMiddleware
+from ..settings import (
+    MOONSTREAM_APPLICATION_ID,
+    DOCS_TARGET_PATH,
+    ORIGINS,
+    DOCS_PATHS,
+    bugout_client as bc,
+)
+from ..version import MOONSTREAM_VERSION
+
+logger = logging.getLogger(__name__)
+
+tags_metadata = [
+    {"name": "users", "description": "Operations with users."},
+    {"name": "tokens", "description": "Operations with user tokens."},
+]
+
+app = FastAPI(
+    title=f"Moonstream users API.",
+    description="User, token and password handlers.",
+    version=MOONSTREAM_VERSION,
+    openapi_tags=tags_metadata,
+    openapi_url="/openapi.json",
+    docs_url=None,
+    redoc_url=f"/{DOCS_TARGET_PATH}",
+)
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=ORIGINS,
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+whitelist_paths: Dict[str, str] = {}
+whitelist_paths.update(DOCS_PATHS)
+whitelist_paths.update(
+    {
+        "/users": "POST",
+        "/users/token": "POST",
+        "/users/password/restore": "POST",
+        "/users/password/reset": "POST",
+    }
+)
+app.add_middleware(BroodAuthMiddleware, whitelist=whitelist_paths)
+
+
+@app.post("/", tags=["users"], response_model=BugoutUser)
```

Comment:
TODO(@kompotkot ): We should allow applications to specify whether they want to send a welcome email on signup. Similarly, verification.

This probably needs a design doc on Brood.